test/libsolidity/util/TestFileParser.cpp

```diff
@@ -70,32 +70,47 @@ vector<dev::solidity::test::FunctionCall> TestFileParser::parseFunctionCalls()
 				if (calls.empty())
 					expect(Token::Newline);
 				else
-					accept(Token::Newline, true);
+					if (accept(Token::Newline, true))
+						m_lineNumber++;
 
-				call.signature = parseFunctionSignature();
-				if (accept(Token::Comma, true))
-					call.value = parseFunctionCallValue();
-				if (accept(Token::Colon, true))
-					call.arguments = parseFunctionCallArguments();
+				try {
 
-				if (accept(Token::Newline, true))
-					call.displayMode = FunctionCall::DisplayMode::MultiLine;
+					call.signature = parseFunctionSignature();
+					if (accept(Token::Comma, true))
+						call.value = parseFunctionCallValue();
+					if (accept(Token::Colon, true))
+						call.arguments = parseFunctionCallArguments();
 
-				call.arguments.comment = parseComment();
+					if (accept(Token::Newline, true))
+					{
+						call.displayMode = FunctionCall::DisplayMode::MultiLine;
+						m_lineNumber++;
+					}
 
-				if (accept(Token::Newline, true))
-					call.displayMode = FunctionCall::DisplayMode::MultiLine;
+					call.arguments.comment = parseComment();
 
-				expect(Token::Arrow);
-				call.expectations = parseFunctionCallExpectations();
+					if (accept(Token::Newline, true))
+					{
+						call.displayMode = FunctionCall::DisplayMode::MultiLine;
+						m_lineNumber++;
+					}
 
-				accept(Token::Newline, true);
-				call.expectations.comment = parseComment();
+					expect(Token::Arrow);
+					call.expectations = parseFunctionCallExpectations();
 
-				if (call.signature == "constructor()")
-					call.isConstructor = true;
+					if (accept(Token::Newline, true))
+						m_lineNumber++;
+					call.expectations.comment = parseComment();
 
-				calls.emplace_back(std::move(call));
+					if (call.signature == "constructor()")
+						call.isConstructor = true;
+
+					calls.emplace_back(std::move(call));
+				}
+				catch (Error const& _e)
+				{
+					throw Error{_e.type(), "Line " + to_string(_lineOffset + m_lineNumber) + ": " + _e.what()};
```

Comment:
Maybe it would be better to use a copy-constructor, because it might copy more of the boost annotations, but could also not be the case...

Also: Why not use the `SourceLocation` parameter? You would have to use offsets instead of line numbers in that case, though.

Reply:
Wouldn't copy construction still lack the line information since we can't add it after? 

Using the `SourceLocation` parameter would be nice, but that needs further tracking of the column which is already lost after tokenization.

Reply:
Right, probably not worth the effort.